libmpv/client.h

```diff
@@ -751,7 +763,8 @@ typedef enum mpv_format {
 typedef struct mpv_node {
     union {
         char *string;   /** valid if format==MPV_FORMAT_STRING */
-        int flag;       /** valid if format==MPV_FORMAT_FLAG   */
+        bool bool_;     /** valid if format==MPV_FORMAT_BOOL or MPV_FORMAT_FLAG */
+        int flag;       /** valid if format==MPV_FORMAT_BOOL or MPV_FORMAT_FLAG   */
```

Comment:
I'm pretty sure the C standard forbids writing to one field and then expecting valid data out of another of an union or something but there's a real case where this won't work: big-endian machines

try this python script:
```py
import struct
for endian in "<>":
	print("endian is", endian == "<" and "little" or "big")
	b = struct.pack(endian + "I", 1)
	print("int(1) is", b)
	b = b[:1] # truncate
	print("int(1) read as bool is", struct.unpack(endian + "B", b)[0])
	b = struct.pack(endian + "B", 1)
	print("bool(1) is", b)
	b += b'\x00\x00\x00' # pad
	print("bool(1) read as int is", struct.unpack(endian + "I", b)[0])
```

Reply:
The intent is for both fields to be written to (which happens, I think..) This comment is just wrong though because the format should always be `MPV_FORMAT_FLAG` since the internally `getproperty_fn` converts `MPV_FORMAT_BOOL` to `MPV_FORMAT_FLAG`.

Reply:
No wait you're right that won't work hmm.

Reply:
I might be completely wrong here, but for testing endian stuff couldn't you set e.g. 0x1000000 (the number that the python script spits out) instead of 1 for flag and see if that will still result in the correct bool?

Reply:
The documentation says "flag" will only ever be 1 or 0. This is impossible to get right.

Reply:
If we ensure that only bool gets stored internally, then we also don't have to deal with reading either flag or bool, since we know for sure it's bool only. Then converting a bool to flag, if that's requested, is trivial independent of endianness.